File: seoyeon/week7/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    keytable = {}
    for i in keymap :
        for idx, j in enumerate(i) :
            if j in keytable :
                keytable[j] = min(keytable[j], idx+1)
            else :
                keytable[j] = idx+1
    for i in targets :
        tmp = 0
        for j in i :
            if j not in keytable:
                tmp = -1
                break
            tmp += keytable[j]
        answer.append(tmp)
    return answer

# https://blog.naver.com/rabbii11/223182609503

File: hyeyoung/week7/대충만든자판.java
import java.util.*;

class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        int[] answer = new int[targets.length];
 
        Map<Character, Integer> map = new HashMap<>();
 
        for (String k: keymap) {
            for (int i = 0; i < k.length(); i++) {
                char key = k.charAt(i);
                map.put(key, Math.min(i + 1, map.getOrDefault(key, Integer.MAX_VALUE)));
            }
        }
 
        for (int i = 0; i < targets.length; i++) {
            String s = targets[i];
 
            for (int j = 0; j < s.length(); j++) {
                char key = s.charAt(j);
 
                if (map.containsKey(key)) {
                    answer[i] += map.get(key);
                } else {
                    answer[i] = -1;
                    break;
                }
            }
        }
 
        return answer;
    }
}
